server/src/weaverbird/backends/sql_translator/steps/ifthenelse.py

```diff
@@ -16,24 +16,44 @@
 from weaverbird.pipeline.steps.ifthenelse import IfThenElse
 
 
-def recursively_convert_nested_condition(step: IfthenelseStep, composed_query: str) -> str:
+def recursively_convert_nested_condition(
+    step: IfthenelseStep,
+    composed_query: str,
+    sql_dialect: SQLDialect,
+) -> str:
     if not hasattr(step, 'else_value'):
         return str(step)
 
+    condition = apply_condition(step.condition, composed_query)
+    expr1 = step.then
+
     if type(step.else_value) != IfThenElse:
-        return (
-            f"""IFF({apply_condition(step.condition, composed_query)}, """
-            f"""{step.then}, {step.else_value})"""
-        )
+        expr2 = step.else_value
+        return build_conditional_expression(
+            sql_dialect, condition, expr1, expr2, last_nested=True
+        ).replace(',)', ')')
     else:
-        composed_query += (
-            f"""IFF({apply_condition(step.condition, composed_query)}, {step.then}, """
-            f"""{recursively_convert_nested_condition(step.else_value, composed_query)})"""
-        )
+        expr2 = recursively_convert_nested_condition(step.else_value, composed_query, sql_dialect)
+        composed_query += build_conditional_expression(sql_dialect, condition, expr1, expr2)
 
     return composed_query.replace(',),', '),')
 
 
+def build_conditional_expression(
+    sql_dialect: SQLDialect, condition, expr1, expr2, last_nested=False
+):
+    if sql_dialect == "postgres":
+        if last_nested:
+            conditional_expression = f"WHEN {condition} THEN {expr1} ELSE {expr2} END"
+        elif last_nested:
+            conditional_expression = f"WHEN {condition} THEN {expr1} ELSE {expr2} END"
+        else:
+            conditional_expression = f"WHEN {condition} THEN {expr1} {expr2}"
+    else:
+        conditional_expression = f"IFF({condition}, {expr1}, {expr2})"
```

Comment:
`IFF` seems to be only implemented by Snowflake while `CASE` seems available in both.
Also, generating the case statement doesn't require nesting, so we could get rid of this recursive method & use an "easier to understand" loop to build it.